### PR TITLE
feat: warn when using --unstable, prefer granular flags

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit deprecates "--unstable" flag. 

When "--unstable" flag is encountered a warning like this is printed:
```
The `--unstable` flag is deprecated, use granular `--unstable-*` flags instead.
Learn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags
```
When "--unstable" flag is used and an unstable API is called an
additional warning like this is printed for each API call:
```
The `Deno.dlopen` API was used with `--unstable` flag. The `--unstable` flag is deprecated, use granular `--unstable-ffi` instead. 
Learn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags
```
When no "--unstable-*" flag is provided and an unstable API is called following
warning is issued before exiting:
```
Unstable API 'Deno.dlopen'. The `--unstable-ffi` flag must be provided.
```

### CODE REVIEW

**Comment by bartlomieju** [comment]
Blocked by https://github.com/denoland/deno_core/pull/400.

**Comment by bartlomieju** [comment]
Tests to fix:
```
# need a `--unstable-jupyter` flag
integration::jupyter::jupyter_install_command_not_exists
# CP.fork uses `--unstable` by default :/
integration::npm::child_process_fork_test
```

File: cli/main.rs [hunk 1]
```diff
@@ -340,16 +340,22 @@ pub(crate) static UNSTABLE_GRANULAR_FLAGS: &[(
...
 
-#[allow(dead_code)]
-pub(crate) fn unstable_warn_cb(feature: &str) {
+// TODO(bartlomieju): remove when `--unstable` flag is removed.
+pub(crate) fn unstable_warn_cb(feature: &str, api_name: &str) {
   eprintln!(
-    "The `--unstable` flag is deprecated, use --unstable-{feature} instead."
+    "⚠️  {}",
+    colors::yellow(format!(
+      "The `{}` API was used with `--unstable` flag. The `--unstable` flag is deprecated, use granular `--unstable-{}` instead.\nLearn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags",
```
Comment (iuioiua) [inline cli/main.rs:356]: Perhaps we should add a note that it'll be removed in Deno v2.

Comment (bartlomieju) [reply cli/main.rs:289]: Yes, that makes sense. @littledivy can do update the message here.

Also the message should say that this flag has no effect when used with `deno check` subcommand.

Comment (bartlomieju) [reply cli/main.rs:289]: Please remove `Use --unsatble* instead`. We no longer conditionally type check - unstable APIs are now always type-checked.

File: cli/main.rs [hunk 2]
```diff
@@ -279,85 +281,22 @@ fn unwrap_or_exit<T>(result: Result<T, AnyError>) -> T {
...
 
-#[allow(dead_code)]
-pub(crate) fn unstable_warn_cb(feature: &str) {
+// TODO(bartlomieju): remove when `--unstable` flag is removed.
+pub(crate) fn unstable_warn_cb(feature: &str, api_name: &str) {
   eprintln!(
-    "The `--unstable` flag is deprecated, use --unstable-{feature} instead."
+    "⚠️  {}",
+    colors::yellow(format!(
+      "The `{}` API was used with `--unstable` flag. The `--unstable` flag is deprecated and will be removed in Deno 2.0, use granular `--unstable-{}` instead.\nLearn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags",
```
Comment (iuioiua) [inline cli/main.rs:297]:
```suggestion
      "The `{}` API was used with `--unstable` flag. The `--unstable` flag is deprecated and will be removed in Deno 2.0. Use granular `--unstable-{}` instead.\nLearn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags",
```

File: cli/main.rs [hunk 3]
```diff
@@ -404,6 +343,25 @@ pub fn main() {
...
+          "⚠️  {}",
+          colors::yellow(
+            "The `--unstable` flag is not needed for `deno check` anymore."
+          )
+        );
+      } else {
+        eprintln!(
+          "⚠️  {}",
+          colors::yellow(
+            "The `--unstable` flag is deprecated and will be removed in Deno 2.0, use granular `--unstable-*` flags instead.\nLearn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags"
```
Comment (iuioiua) [inline cli/main.rs:359]:
```suggestion
            "The `--unstable` flag is deprecated and will be removed in Deno 2.0. Use granular `--unstable-*` flags instead.\nLearn more at: https://docs.deno.com/runtime/manual/tools/unstable_flags"
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Nits

**Review by littledivy — APPROVED** [review]
LGTM!

**Pull request merged**